File: python/trees/tree.py
from BinaryTree import *

#          F
#        /   \
#      B       G
#    /   \       \
#  A      D       I
#       /   \    /
#      C     E  H

a = BinaryTree('A')
b = BinaryTree('B')
c = BinaryTree('C')
d = BinaryTree('D')
e = BinaryTree('E')
f = BinaryTree('F')
g = BinaryTree('G')
h = BinaryTree('H')
i = BinaryTree('I')

f.left  = b
f.right = g
b.left  = a
b.right = d
d.left  = c
d.right = e
g.right = i
i.left  = h

depthFirstPreOrder(f)
print

depthFirstInOrder(f)
print

depthFirstPostOrder(f)
print

File: scala/tree.scala
object Tree {
/*
        F
      /   \
    B       G
  /   \       \
A      D       I
     /   \    /
    C     E  H

*/

  def main(args: Array[String]) {
    var a = new BinaryTreeNode("A");
    var b = new BinaryTreeNode("B");
    var c = new BinaryTreeNode("C");
    var d = new BinaryTreeNode("D");
    var e = new BinaryTreeNode("E");
    var f = new BinaryTreeNode("F");
    var g = new BinaryTreeNode("G");
    var h = new BinaryTreeNode("H");
    var i = new BinaryTreeNode("I");

    f.left  = b
    f.right = g
    b.left  = a
    b.right = d
    d.left  = c
    d.right = e
    g.right = i
    i.left  = h

    var t = new BinaryTree(f);
    t.depthFirstInOrderPrint();
    println();
    t.depthFirstPreOrderPrint();
    println();
    t.depthFirstPostOrderPrint();
    println();
    println();
    t.breadthFirstLevelOrderPrint();
  }
}
